overlord/snapstate/sequence/sequence.go

```diff
@@ -30,19 +30,30 @@ import (
 	"github.com/snapcore/snapd/snap/naming"
 )
 
+// ComponentState contains information about an installed component.
+type ComponentState struct {
+	SideInfo *snap.ComponentSideInfo `json:"side-info"`
+	CompType snap.ComponentType      `json:"type"`
+}
+
+// NewComponentState creates a ComponentState from components side information and type.
+func NewComponentState(si *snap.ComponentSideInfo, tp snap.ComponentType) *ComponentState {
+	return &ComponentState{SideInfo: si, CompType: tp}
+}
+
 // RevisionSideState contains the side information for a snap and related components
 // installed in the system.
 type RevisionSideState struct {
 	Snap       *snap.SideInfo
-	Components []*snap.ComponentSideInfo
+	Components []*ComponentState
 }
 
 // revisionSideInfoMarshal is an ancillary structure used exclusively to
 // help marshaling of RevisionSideInfo.
 type revisionSideInfoMarshal struct {
 	// SideInfo is included for compatibility with older snapd state files
 	*snap.SideInfo
-	CompSideInfo []*snap.ComponentSideInfo `yaml:"components,omitempty" json:"components,omitempty"`
+	CompSideInfo []*ComponentState `json:"components,omitempty"`
```

Comment:
should we change the struct field name here? I don't have a storng opinion myself

Reply:
Changed it now, this was actually an overlook